### ISSUE TITLE
XGB Ranker predicts 0.5 for all data points 

### ISSUE THREAD

**Issue by pearlli98** [dmlc/xgboost#6702]
Problem: 
XGB ranker fitted to data and labels only predicts 0.5

How to reproduce:
```
#generate random training data and labels (range 1-4, inclusive)
train = pd.DataFrame(np.random.randint(0,100,size=(100, 4)), columns=list('ABCD'))
train['label'] = np.random.randint(1, 5, (100))
#sort dataframe by label in ascending order
train = train.sort_values(by = ['label'])
#drop label to generate training data
dtrain = train.drop('label', axis=1)
#get array of value counts 
occurCount = train['label'].value_counts().sort_index()

#train XGB Ranker 
xgb_ranker_params = {
    'objective': 'rank:pairwise',
}
ranker = xgb.XGBRanker(**xgb_ranker_params)
sc = StandardScaler()
ranker.fit(sc.fit_transform(dtrain), train['label'], group = occurCount)

#generate random testing data
test = pd.DataFrame(np.random.randint(0,100,size=(20, 4)), columns=list('ABCD'))
print(ranker.predict(sc.transform(test)))
```

Output: 
`[0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5
 0.5 0.5]`

Parameters of ranker printed out:
`{'objective': 'rank:pairwise',
 'base_score': 0.5,
 'booster': 'gbtree',
 'colsample_bylevel': 1,
 'colsample_bynode': 1,
 'colsample_bytree': 1,
 'gamma': 0,
 'gpu_id': -1,
 'importance_type': 'gain',
 'interaction_constraints': '',
 'learning_rate': 0.300000012,
 'max_delta_step': 0,
 'max_depth': 6,
 'min_child_weight': 1,
 'missing': nan,
 'monotone_constraints': '()',
 'n_estimators': 100,
 'n_jobs': 16,
 'num_parallel_tree': 1,
 'random_state': 0,
 'reg_alpha': 0,
 'reg_lambda': 1,
 'scale_pos_weight': None,
 'subsample': 1,
 'tree_method': 'exact',
 'validate_parameters': 1,
 'verbosity': None}`

**Comment by PascalIversen** [comment]
Hey,
your relevance labels are constant for every group/query, therefore your predictions are also constant.
I am not sure what exactly you are trying to do, but here is an example which would work:
```python
# creating dummy ranking data
import numpy as np
groups = [60, 50, 32, 22, 22 , 22 , 20, 12]
labels = []
features = []
for group in groups:
  label_group = list(np.argsort(np.argsort(np.random.randn(group))))
  labels = labels + label_group
  for label in label_group:
    features.append([label+10, np.random.randn(), np.random.randn()]) 
features = np.stack(features)


# fitting a ranking model

import xgboost as xgb
import pandas

model = xgb.XGBRanker(
        objective="rank:pairwise",
        max_depth= 10,
        n_estimators=1000,
         verbosity=2)

ranker = model.fit(X=features,
        y=labels, group=groups)
model.predict(features)
```
